src/services/post.service.ts

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { Action, ActionTypes, NotificationTypes } from '../reducer/store';
+import { getPostComments, getUserPost } from '../api/users.api';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+export function postService(
+  dispatch: (action: Action) => void,
+) {
+  function setPosts(
+    setShowNotification: React.Dispatch<React.SetStateAction<boolean>>,
+    selectedUser: User | null,
+    setShowSpinner: React.Dispatch<React.SetStateAction<boolean>>,
+  ) {
+    setShowNotification(false);
+
+    dispatch(
+      { type: ActionTypes.selectPost, post: null },
+    );
+
+    if (selectedUser) {
+      setShowSpinner(true);
+
+      dispatch(
+        { type: ActionTypes.getAllPosts, posts: null },
+      );
+
+      getUserPost(selectedUser.id)
+        .then(posts => {
+          if (!Array.isArray(posts)) {
+            throw Error();
+          }
+
+          if (posts.length === 0) {
```

Comment:
This line is dispatching 'getAllPosts' action with posts set to null which is the same action dispatched in line 24. It's redundant and can be removed.